sdk/security/keyvault/azadmin/backup/client_test.go

```diff
@@ -177,3 +181,48 @@ func TestBeginSelectiveKeyRestoreOperation(t *testing.T) {
 		time.Sleep(60 * time.Second)
 	}
 }
+
+func TestAPIVersion(t *testing.T) {
+	apiVersion := "7.3"
+	var requireVersion = func(t *testing.T) func(req *http.Request) bool {
+		return func(r *http.Request) bool {
+			version := r.URL.Query().Get("api-version")
+			require.Equal(t, version, apiVersion)
+			return true
+		}
+	}
+	srv, close := mock.NewServer(mock.WithTransformAllRequestsToTestServerUrl())
+	defer close()
+	srv.AppendResponse(
+		mock.WithHeader("WWW-Authenticate", `Bearer authorization="https://login.microsoftonline.com/tenant", resource="https://managedhsm.azure.net"`),
+		mock.WithStatusCode(401),
+		mock.WithPredicate(requireVersion(t)),
+	)
+	srv.AppendResponse() // when a response's predicate returns true, srv pops the following one
```

Comment:
Just occurred to me this is a better pattern because it makes the test fail if `srv` returns the response we expect it to skip
```suggestion
	srv.AppendResponse(mock.WithStatusCode(http.StatusInternalServerError))
```